.github/workflows/update-contribs.yml

```diff
@@ -1,46 +1,45 @@
 jobs:
-    run-meta:
-        runs-on: ubuntu-latest
-        steps:
-        - name: Check out the code
-          uses: actions/checkout@v3
-        - name: Setup Python
-          uses: actions/setup-python@v4
-          with:
-            python-version: '3.10'
-        - name: Upgrade pip
-          run: |
-            # install pip=>20.1 to use "pip cache dir"
-            python -m pip install --upgrade pip wheel
-            echo "$PWD"
-            git branch
-        - name: Install pyosmeta and run update contribs
-          env:
-            GITHUB_TOKEN: ${{ github.token }}
-          run: |
-            pip install pyosmeta@git+https://github.com/pyopensci/update-web-metadata
-            echo "$PWD"
-            # Skip error for when this pr actually works
-            # python -bb -W error scripts/parse-contributors.py
-            python scripts/parse-contributors.py
-        - name: Create Pull Request
-          uses: peter-evans/create-pull-request@v5
-          with:
-            add-paths: _data/contributors.yml
-            author: Leah <leah@pyopensci.org>
-            base: main
-            branch: contribs
-            commit-message: "Update: Contributor file update"
-            delete-branch: true
-            title: Update new contributors
-          env:
-            # Token needed to provide permissions to bot to push to branch
-            # https://github.com/peter-evans/create-pull-request/issues/48#issuecomment-536204092
-            GITHUB_TOKEN: ${{ secrets.PYOS_PR_TOKEN }}
+  run-meta:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the code
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Upgrade pip
+        run: |
+          # install pip=>20.1 to use "pip cache dir"
+          python -m pip install --upgrade pip wheel
+          echo "$PWD"
+          git branch
+      - name: Install pyosmeta and run update contribs
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          pip install pyosmeta@git+https://github.com/pyopensci/update-web-metadata
+          echo "$PWD"
+          # Skip error for when this pr actually works
+          # python -bb -W error scripts/parse-contributors.py
+          python scripts/parse-contributors.py
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v5
+        with:
+          add-paths: _data/contributors.yml
+          author: Leah <leah@pyopensci.org>
+          base: main
+          branch: contribs
+          commit-message: "Update: Contributor file update"
+          delete-branch: true
+          title: Update new contributors
+        env:
+          # Token needed to provide permissions to bot to push to branch
+          # https://github.com/peter-evans/create-pull-request/issues/48#issuecomment-536204092
+          GITHUB_TOKEN: ${{ secrets.PYOS_PR_TOKEN }}
```

Comment:
```suggestion
          # Custom token needed to trigger PR checks, as GITHUB_TOKEN won't
          # https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs
          GITHUB_TOKEN: ${{ secrets.PYOS_PR_TOKEN }}
```

Upon checking the comment that you linked, I see what the actual issue is—as the linked comment states (which was later updated with a link to the [relevant canonical section](https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs) in the pull request action docs), the token is _not_ required to push to a branch (that just requires setting the workflow `permissions` correctly), but rather is necessary to get GitHub to trigger the automatic PR checks on the PR it opens, without manually triggering them action (e.g. closing and re-opening it), due to [GitHub not triggering other actions based on an action](https://github.com/orgs/community/discussions/35047).

Therefore, I've updated the comment accordingly.

Also, just to make sure—you are using a new limited, repo-scoped access token, right? Otherwise, anyone who gets access to the token could compromise your whole account.

Reply:
yes i scoped the token just to this repo in this organization only and just for this PR purpose if that makes sense. it seemed like i did need that to create an actual pull request.

Reply:
> it seemed like i did need that to create an actual pull request.

As explained previously, for that you just needed to enable the `pull requests: write` permission for `GITHUB_TOKEN` with

```
permissions:
    pull-request: write
```

The PAT is just recommended because of the above-described issue where using `GITHUB_TOKEN` for opening PRs doesn't automatically trigger the CI checks on the PR, requiring them to be triggered manually, so using it avoids that inconvenience.